ui/media/js/dnd.js

```diff
@@ -306,19 +313,25 @@ function restoreTaskToUI(task, fieldsToSkip) {
     }
     if (!('mask' in task.reqBody)) {
         maskSetting.checked = false
+        maskSetting.dispatchEvent(new Event("click"))
     }
     upscaleModelField.disabled = !useUpscalingField.checked
     upscaleAmountField.disabled = !useUpscalingField.checked
 
     // Show the source picture if present
     initImagePreview.src = (task.reqBody.init_image == undefined ? '' : task.reqBody.init_image)
     if (IMAGE_REGEX.test(initImagePreview.src)) {
+        initImagePreview.dispatchEvent(new Event("load"))
         if (Boolean(task.reqBody.mask)) {
             setTimeout(() => { // add a delay to insure this happens AFTER the main image loads (which reloads the inpainter)
                 imageInpainter.setImg(task.reqBody.mask)
             }, 250)
         }
     }
+    else
+    {
+        initImageClearBtn.dispatchEvent(new Event("click"))
```

Comment:
Shouldn't this "click" event only be dispatched if the image contained something (before we set it to empty in line 322)? Otherwise this isn't a state change.

Maybe by checking the regex before we set the image src (i.e. before line 322) to determine whether the image was empty before we set something.

Reply:
Fixed, and also replaced the hardcoded delay with a smoother and more robust event-driven flow.